src/Orleans/Messaging/BufferPool.cs

```diff
@@ -1,20 +1,25 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Threading;
 using Orleans.Runtime.Configuration;
 
 namespace Orleans.Runtime
 {
     internal class BufferPool
     {
         private readonly int byteBufferSize;
-        private readonly BlockingCollection<byte[]> buffers;
+        private readonly int maxBuffersCount;
+        private readonly bool limitBuffersCount;
+        private readonly ConcurrentBag<byte[]> buffers;
         private readonly CounterStatistic allocatedBufferCounter;
         private readonly CounterStatistic checkedOutBufferCounter;
         private readonly CounterStatistic checkedInBufferCounter;
         private readonly CounterStatistic droppedBufferCounter;
         private readonly CounterStatistic droppedTooLargeBufferCounter;
 
+        private int currentAllocatedBuffers;
```

Comment:
currentAllocatedBuffers, to me seems an unclear name.  Since this is related to maxBufferCount and limitBufferCount, maybe currentBufferCount is a more clear name?